### PR TITLE
Fixes StringIndexOutOfBoundsException when cutting text

### LINKED ISSUE #5708: Cutting some of the bibtexkey raises error: Bale13 -> Bal13
JabRef 5.0.0-dev--2019-11-28----ce0e8870c
Windows 7 6.1 amd64 
Java 13.0.1

- [ ] I have tested the latest development version from http://builds.jabref.org/master/ and the problem persists

1. I imported an entry via zotero (since jabfox doesnot work
2. the resulting bibtexkey has been e.g. Bale13 or Heinrich87
3. These I reduce to Bal13 or Hei87. 
4. Immediately an java error is shown.
5. Once the error is clicked away, a second attempt to change bixtexkey works flawlessly.

<details>
  <summary>Log File</summary>
  
  ```
  java.lang.StringIndexOutOfBoundsException: begin 3, end 8, length 5
	at java.base/java.lang.String.checkBoundsBeginEnd(Unknown Source)
	at java.base/java.lang.String.substring(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl$2.computeValue(Unknown Source)
	at org.jabref.merged.module/javafx.beans.binding.StringBinding.get(Unknown Source)
	at org.jabref.merged.module/javafx.beans.binding.StringBinding.get(Unknown Source)
	at org.jabref.merged.module/javafx.beans.binding.StringExpression.getValue(Unknown Source)
	at org.jabref.merged.module/javafx.beans.binding.StringExpression.getValue(Unknown Source)
	at org.jabref.merged.module/javafx.beans.property.StringPropertyBase.get(Unknown Source)
	at org.jabref.merged.module/javafx.beans.property.ReadOnlyStringWrapper$ReadOnlyPropertyImpl.get(Unknown Source)
	at org.jabref.merged.module/javafx.beans.property.ReadOnlyStringWrapper$ReadOnlyPropertyImpl.get(Unknown Source)
	at org.jabref.merged.module/javafx.beans.binding.StringExpression.getValue(Unknown Source)
	at org.jabref.merged.module/javafx.beans.binding.StringExpression.getValue(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper$SingleChange.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/javafx.beans.property.ReadOnlyStringPropertyBase.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/javafx.beans.property.ReadOnlyStringWrapper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/javafx.beans.property.StringPropertyBase.markInvalid(Unknown Source)
	at org.jabref.merged.module/javafx.beans.property.StringPropertyBase$Listener.invalidated(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper$SingleInvalidation.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/javafx.beans.binding.StringBinding.invalidate(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.binding.BindingHelperObserver.invalidated(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper$Generic.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl$TextProperty.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl$TextProperty.markInvalid(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl$TextProperty.controlContentHasChanged(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl.lambda$new$0(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper$SingleInvalidation.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.binding.ExpressionHelper.fireValueChangedEvent(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextField$TextFieldContent.delete(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl.replaceText(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl.updateContent(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl.replaceText(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl.replaceText(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl.replaceSelection(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.TextInputControl.deleteNextChar(Unknown Source)
	at org.jabref.merged.module/javafx.scene.control.skin.TextFieldSkin.deleteChar(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.scene.control.behavior.TextFieldBehavior.deleteChar(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.scene.control.behavior.TextInputControlBehavior.deleteNextChar(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.scene.control.behavior.TextInputControlBehavior.lambda$new$14(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.scene.control.behavior.TextInputControlBehavior.lambda$keyMapping$62(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.scene.control.inputmap.InputMap.handle(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at org.jabref.merged.module/javafx.event.Event.fireEvent(Unknown Source)
	at org.jabref.merged.module/javafx.scene.Scene$KeyHandler.process(Unknown Source)
	at org.jabref.merged.module/javafx.scene.Scene.processKeyEvent(Unknown Source)
	at org.jabref.merged.module/javafx.scene.Scene$ScenePeerListener.keyEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.tk.quantum.GlassViewEventHandler$KeyEventNotification.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleKeyEvent$1(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(Unknown Source)
	at org.jabref.merged.module/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleKeyEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.glass.ui.View.handleKeyEvent(Unknown Source)
	at org.jabref.merged.module/com.sun.glass.ui.View.notifyKey(Unknown Source)
	at org.jabref.merged.module/com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at org.jabref.merged.module/com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
  ```
</details>

### PR DESCRIPTION
**Pull request by tobiasdiez**
There is a  bug in the jdk that adding a listener to the `selectedTextProperty` leads to StringIndexOutOfBoundsException when cutting / deleting text from the field: https://bugs.openjdk.java.net/browse/JDK-8176270

This is the reason for #5708. I fixed it by uncommenting the problematic code, which was related to the X11 support. @davidemdot @koppor How important is this code for X11? Is there a workaround that does not add a listener to `selectedTextProperty`?

- [ ] Change in CHANGELOG.md described (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for bigger UI changes)
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not: Issue created at <https://github.com/JabRef/user-documentation/issues>.

Fixes #5708.

### CODE REVIEW

File: src/main/java/org/jabref/gui/ClipBoardManager.java [hunk 1]
```diff
@@ -66,11 +66,13 @@ public ClipBoardManager(Clipboard clipboard, java.awt.datatransfer.Clipboard pri
      * @see <a href="https://unix.stackexchange.com/questions/139191/whats-the-difference-between-primary-selection-and-clipboard-buffer/139193#139193">Longer text over clipboards</a>
      */
     public static void addX11Support(TextInputControl input) {
+        /*
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/ClipBoardManager.java]: Please add a link to the jdk bug, so we can check if it's fixed later on.

**Comment by koppor** [comment]
The code adds support for X11 primary selection clipboard (see https://github.com/JabRef/jabref/pull/5482). This is essential for advanced linux users using the hidden powers of their system (and not migrating from Windows and thus not using things what Unix offers since 40 years).

Would it be possible to fix the code somewhere in JavaFX? - The code is open sourced: https://github.com/openjdk/jfx (hint by  @dlemmermann). Refs my rant against JavaFX at https://twitter.com/koppor/status/1181926475957706753. That rant was too harsh, I know, but I did not expect these kind of issues (as also this one).

**Comment by koppor** [comment]
The linked JDK issue presents following workaround code. Maybe, this can be done here, too?

> use `InvalidationListener` instead of `ChangeListener` and encapsulate access to `selectedTextProperty` in `Platform.runLater` (see provided test cases)

**Comment by koppor** [comment]
Just as an impression how stable the code in TextInputControl is:

grafik

**Comment by tobiasdiez** [comment]
@koppor can you please take over this PR and try to implement the workaround. I don't have a Linux available to test it.

**Comment by koppor** [comment]
Working on it

grafik

**Comment by koppor** [comment]
Ported the fix. Need to fire up my [vcxsrv](https://sourceforge.net/projects/vcxsrv/) in my [WSL](https://docs.microsoft.com/de-de/windows/wsl/install-win10) as described at https://medium.com/@dhanar.santika/installing-wsl-with-gui-using-vcxsrv-6f307e96fac0.

**Comment by tobiasdiez** [comment]
Cool! I guess we still need to use the above workaround until your fix is in a published JavaFX release.

**Comment by koppor** [comment]
Checked the fix. Works on Linux and Mac.

**Pull request merged**